setup/classes/class.ilCtrlStructureReader.php

```diff
@@ -102,146 +102,189 @@ function readStructure($a_force = false, $a_dir = "", $a_comp_prefix = "",
 	function read($a_cdir)
 	{
 		$ilDB = $this->getDB();
-		$il_absolute_path = realpath(dirname(__FILE__) .'/../../');
+		$il_absolute_path = realpath(dirname(__FILE__) . '/../../');
 
-		// check wether $a_cdir is a directory
-		if (!@is_dir($a_cdir))
-		{
+		$a_cdir = preg_replace('#//#', '/', $a_cdir);
```

Comment:
This seems like the code that passes $a_cdir makes something wrong. Is the cause of the problem somewhere else?